pkg/substitution/substitution_test.go

```diff
@@ -116,7 +117,196 @@ func TestSetCloudControllerImage(t *testing.T) {
+			ClusterProxy: &configv1.Proxy{
+				Status: configv1.ProxyStatus{
+					HTTPProxy: "http://squid.corp.acme.com:3128",
+				},
+			},
+		},
+	}, {
+		name: "Substitute https proxy",
+		containers: []corev1.Container{{
+			Name: "https_proxy",
+			Env: []corev1.EnvVar{{
+				Name:  "SomeVar",
+				Value: "SomeValue",
+			}},
+		}},
+		expectedContainers: []corev1.Container{{
+			Name: "https_proxy",
+			Env: []corev1.EnvVar{
+				{
+					Name:  "SomeVar",
+					Value: "SomeValue",
+				}, {
+					Name:  "HTTPS_PROXY",
+					Value: "https://squid.corp.acme.com:3128",
+				}},
+		}},
+		config: config.OperatorConfig{
+			ClusterProxy: &configv1.Proxy{
+				Status: configv1.ProxyStatus{
+					HTTPSProxy: "https://squid.corp.acme.com:3128",
+				},
+			},
+		},
+	}, {
+		name: "Substitute no proxy",
+		containers: []corev1.Container{{
+			Name: "no_proxy",
+			Env: []corev1.EnvVar{{
+				Name:  "SomeVar",
+				Value: "SomeValue",
+			}},
+		}},
+		expectedContainers: []corev1.Container{{
+			Name: "no_proxy",
+			Env: []corev1.EnvVar{
+				{
+					Name:  "SomeVar",
+					Value: "SomeValue",
+				}, {
+					Name:  "NO_PROXY",
+					Value: "https://internal.acme.com",
+				}},
+		}},
+		config: config.OperatorConfig{
+			ClusterProxy: &configv1.Proxy{
+				Status: configv1.ProxyStatus{
+					NoProxy: "https://internal.acme.com",
+				},
+			},
+		},
+	}, {
+		name: "Combination of proxy settings",
+		containers: []corev1.Container{{
+			Name: "all_proxy",
+			Env: []corev1.EnvVar{{
+				Name:  "SomeVar",
+				Value: "SomeValue",
+			}},
+		}},
+		expectedContainers: []corev1.Container{{
+			Name: "all_proxy",
+			Env: []corev1.EnvVar{{
+				Name:  "SomeVar",
+				Value: "SomeValue",
+			}, {
+				Name:  "HTTP_PROXY",
+				Value: "http://squid.corp.acme.com:3128",
+			}, {
+				Name:  "HTTPS_PROXY",
+				Value: "https://squid.corp.acme.com:3128",
+			}, {
+				Name:  "NO_PROXY",
+				Value: "https://internal.acme.com",
+			}},
+		}},
+		config: config.OperatorConfig{
+			ClusterProxy: &configv1.Proxy{
+				Status: configv1.ProxyStatus{
+					HTTPProxy:  "http://squid.corp.acme.com:3128",
+					HTTPSProxy: "https://squid.corp.acme.com:3128",
+					NoProxy:    "https://internal.acme.com",
+				},
+			},
+		},
+	}}
+
+	for _, tc := range tc {
+		t.Run(tc.name, func(t *testing.T) {
+			deploy := &v1.Deployment{
+				Spec: v1.DeploymentSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: tc.containers,
+						},
+					},
+				},
+			}
+
+			ds := &v1.DaemonSet{
+				Spec: v1.DaemonSetSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: tc.containers,
+						},
+					},
+				},
+			}
+
+			spec := setProxySettings(tc.config, deploy.Spec.Template.Spec)
+			assert.EqualValues(t, spec.Containers, tc.expectedContainers)
+
+			spec = setProxySettings(tc.config, ds.Spec.Template.Spec)
+			assert.EqualValues(t, spec.Containers, tc.expectedContainers)
```

Comment:
I don't think there's a need to have the deployment or daemonset in this, nothing specific about them is being observed here is it? I think we can just pass the podSpec and run this once